diff -uNr a/build/linux/dist/arduino b/build/linux/dist/arduino
--- a/build/linux/dist/arduino	2015-11-07 16:21:27.391410539 +0100
+++ b/build/linux/dist/arduino	2015-11-07 16:22:19.534411671 +0100
@@ -8,11 +8,11 @@
     "$APPDIR"/lib/*.jar \
     ;
 do
-    CLASSPATH="${CLASSPATH}:${LIB}"
+    CLASSPATH="${CLASSPATH+$CLASSPATH:}${LIB}"
 done
 export CLASSPATH
 
-LD_LIBRARY_PATH=$APPDIR/lib${LD_LIBRARY_PATH:+:$LD_LIBRARY_PATH}
+LD_LIBRARY_PATH="$(java-config -di rxtx-2)${LD_LIBRARY_PATH+:$LD_LIBRARY_PATH}"
 export LD_LIBRARY_PATH
 
 export PATH="${APPDIR}/java/bin:${PATH}"
