diff --git src/drmmode_display.c src/drmmode_display.c
index 57533ee..51aee29 100644
--- src/drmmode_display.c
+++ src/drmmode_display.c
@@ -59,7 +59,7 @@
 #include "mipointer.h"
 
 /* All drivers implementing backing store need this */
-#include "mibstore.h"
+//#include "mibstore.h"
 
 #include "micmap.h"
 
diff --git src/omap_driver.h src/omap_driver.h
index 8d49cf6..cfb5a57 100644
--- src/omap_driver.h
+++ src/omap_driver.h
@@ -40,7 +40,7 @@
  * XXX - figure out what can be removed..
  */
 #include "mipointer.h"
-#include "mibstore.h"
+//#include "mibstore.h"
 #include "micmap.h"
 #include "colormapst.h"
 #include "xf86cmap.h"
