--- src/omap_driver.c.orig	2013-08-05 22:06:21.205338459 -0700
+++ src/omap_driver.c	2013-08-05 22:08:06.660338663 -0700
@@ -759,8 +759,8 @@
 	 */
 	OMAPAccelInit(pScreen);
 
-	/* Initialize backing store: */
-	miInitializeBackingStore(pScreen);
+	/* Initialize backing store:
+	miInitializeBackingStore(pScreen); */
 	xf86SetBackingStore(pScreen);
 
 	/* Cause the cursor position to be updated by the mouse signal handler: */
