--- src/mtpz.c.orig	2015-09-16 17:20:36.096068147 -0700
+++ src/mtpz.c	2015-09-16 17:28:04.313541977 -0700
@@ -256,7 +256,7 @@
 mtpz_rsa_t *mtpz_rsa_init(const unsigned char *str_modulus, const unsigned char *str_privkey, const unsigned char *str_pubexp)
 {
 	mtpz_rsa_t *rsa = (mtpz_rsa_t *)malloc(sizeof(mtpz_rsa_t));
-	memset(rsa, 0, sizeof(rsa));
+	memset(rsa, 0, sizeof(mtpz_rsa_t));
 
 	gcry_mpi_t mpi_modulus, mpi_privkey, mpi_pubexp;
 
