--- Makefile.orig	2010-05-26 20:10:07.000000000 -0700
+++ Makefile	2010-08-27 20:50:14.813190580 -0700
@@ -1,6 +1,6 @@
 .PHONY: clean install tests build
 
-prefix ?= /usr/local
+prefix ?= /usr
 bindir ?= $(prefix)/bin
 datadir ?= $(prefix)/share
 
@@ -32,12 +32,12 @@
 	mkdir -p $(DESTDIR)$(datadir)/applications
 	$(INSTALL_DATA) bleachbit.desktop $(DESTDIR)$(datadir)/applications/
 
-	# Python code
-	mkdir -p $(DESTDIR)$(datadir)/bleachbit
-	$(INSTALL_DATA) bleachbit/*.py $(DESTDIR)$(datadir)/bleachbit
-	cd $(DESTDIR)$(datadir)/bleachbit && \
-	python -O -c "import compileall; compileall.compile_dir('.')" && \
-	python -c "import compileall; compileall.compile_dir('.')"
+#	# Python code
+#	mkdir -p $(DESTDIR)$(datadir)/bleachbit
+#	$(INSTALL_DATA) bleachbit/*.py $(DESTDIR)$(datadir)/bleachbit
+#	cd $(DESTDIR)$(datadir)/bleachbit && \
+#	python -O -c "import compileall; compileall.compile_dir('.')" && \
+#	python -c "import compileall; compileall.compile_dir('.')"
 
 	# cleaners
 	mkdir -p $(DESTDIR)$(datadir)/bleachbit/cleaners
