--- camlibs/ptp2/library.c.orig	2011-02-27 13:08:49.000000000 -0800
+++ camlibs/ptp2/library.c	2011-02-27 13:12:15.000000000 -0800
@@ -710,6 +710,8 @@
 	{"Nikon:Coolpix L20 (PTP mode)",  0x04b0, 0x0317, PTP_CAP|PTP_NIKON_BROKEN_CAP},
 	/* https://sourceforge.net/tracker/index.php?func=detail&aid=2947644&group_id=8874&atid=108874 */
 	{"Nikon:Coolpix L19 (PTP mode)",  0x04b0, 0x0318, PTP_CAP|PTP_NIKON_BROKEN_CAP},
+	/* Maria's camera is missing... */
+	{"Nikon:Coolpix L22 (PTP mode)",  0x04b0, 0x031c, PTP_CAP|PTP_NIKON_BROKEN_CAP},
 	/* Nikon D100 has a PTP mode: westin 2002.10.16 */
 	{"Nikon:DSC D100 (PTP mode)",     0x04b0, 0x0402, 0},
 	/* D2H SLR in PTP mode from Steve Drew <stevedrew@gmail.com> */
