--- doxygen-1.7.5.1.ebuild	2011-11-05 21:30:54.000000000 +0000
+++ doxygen-1.7.6.1.ebuild	2011-12-16 10:01:45.402292449 +0000
@@ -36,9 +36,13 @@ src_prepare() {
 	# use CFLAGS, CXXFLAGS, LDFLAGS
 	export ECFLAGS="${CFLAGS}" ECXXFLAGS="${CXXFLAGS}" ELDFLAGS="${LDFLAGS}"
 
-	sed -i.orig -e 's:^\(TMAKE_CFLAGS_RELEASE\t*\)= .*$:\1= $(ECFLAGS):' \
-		-e 's:^\(TMAKE_CXXFLAGS_RELEASE\t*\)= .*$:\1= $(ECXXFLAGS):' \
-		-e 's:^\(TMAKE_LFLAGS_RELEASE\s*\)=.*$:\1= $(ELDFLAGS):' \
+	sed -i.orig -e 's:^\(TMAKE_CFLAGS_\(RELEASE\|DEBUG\)\s*\)=.*$:\1= $(ECFLAGS):' \
+		-e 's:^\(TMAKE_CXXFLAGS_\(RELEASE\|DEBUG\)\s*\)=.*$:\1= $(ECXXFLAGS):' \
+		-e 's:^\(TMAKE_LFLAGS_\(RELEASE\|DEBUG\)\s*\)=.*$:\1= $(ELDFLAGS):' \
+		-e "s:^\(TMAKE_CC\s*=\).*$:\1 $(tc-getCC):g" \
+		-e "s:^\(TMAKE_CXX\s*=\).*$:\1 $(tc-getCXX):g" \
+		-e "s:^\(TMAKE_LINK\s*=\).*$:\1 $(tc-getCXX):g" \
+		-e "/^TMAKE_AR\s*=/s:\<ar\>:$(tc-getAR):g" \
 		tmake/lib/{{linux,gnu,freebsd,netbsd,openbsd,solaris}-g++,macosx-c++,linux-64}/tmake.conf \
 		|| die "sed 1 failed"
 
@@ -50,7 +54,7 @@ src_prepare() {
 	fi
 
 	# Call dot with -Teps instead of -Tps for EPS generation - bug #282150
-	epatch "${FILESDIR}"/${P}-dot-eps.patch
+	epatch "${FILESDIR}"/${PN}-1.7.5.1-dot-eps.patch
 
 	# prefix search tools patch, plus OSX fixes
 	epatch "${FILESDIR}"/${PN}-1.5.6-prefix-misc-alt.patch
@@ -72,6 +76,22 @@ src_prepare() {
 		echo
 		replace-flags "-O3" "-O2"
 	fi
+
+	if use qt4; then
+		# doxywizard built without respecting LDFLAGS,
+		# so add eqmake4 wrapper flags here.
+		cat >> "${S}/addon/doxywizard/doxywizard.pro.in" <<-EOF || die
+			QMAKE_CC    = $(tc-getCC)
+			QMAKE_CXX   = $(tc-getCXX)
+			QMAKE_LINK  = $(tc-getCXX)
+			QMAKE_CFLAGS_RELEASE    += ${CFLAGS}
+			QMAKE_CFLAGS_DEBUG      += ${CFLAGS}
+			QMAKE_CXXFLAGS_RELEASE  += ${CXXFLAGS}
+			QMAKE_CXXFLAGS_DEBUG    += ${CXXFLAGS}
+			QMAKE_LFLAGS_RELEASE    += ${LDFLAGS}
+			QMAKE_LFLAGS_DEBUG      += ${LDFLAGS}
+		EOF
+	fi
 }
 
 src_configure() {
@@ -89,24 +109,12 @@ src_configure() {
 
 	use qt4 && my_conf="${my_conf} --with-doxywizard"
 
-	export CC="${QMAKE_CC}"
-	export CXX="${QMAKE_CXX}"
-	export LINK="${QMAKE_LINK}"
-	export LINK_SHLIB="${QMAKE_CXX}"
-
 	./configure --prefix "${EPREFIX}/usr" ${my_conf} \
 			|| die 'configure failed'
 }
 
 src_compile() {
-
-	# force stupid qmake to behave - if it works...
-	if use qt4 ; then
-		qt4-r2_src_compile
-	else
-		CFLAGS+="${ECFLAGS}" CXXFLAGS+="${ECXXFLAGS}" LFLAGS+="${ELDFLAGS}" \
-			emake all || die 'emake failed'
-	fi
+	emake all || die 'emake failed'
 
 	# generate html and pdf (if tetex in use) documents.
 	# errors here are not considered fatal, hence the ewarn message
@@ -138,7 +146,7 @@ src_compile() {
 }
 
 src_install() {
-	make DESTDIR="${D}" MAN1DIR=share/man/man1 \
+	emake DESTDIR="${ED}" MAN1DIR=share/man/man1 \
 		install || die '"make install" failed.'
 
 	if use qt4; then
