Index: src/mpi/romio/adio/ad_pvfs2/ad_pvfs2.h
===================================================================
--- src/mpi/romio/adio/ad_pvfs2/ad_pvfs2.h	(revision 3699)
+++ src/mpi/romio/adio/ad_pvfs2/ad_pvfs2.h	(revision 3700)
@@ -13,6 +13,10 @@
 #include "pvfs2.h"
 #endif
 
+#ifdef PVFS2_VERSION_MAJOR
+#include "pvfs2-compat.h"
+#endif
+
 void ADIOI_PVFS2_Open(ADIO_File fd, int *error_code);
 void ADIOI_PVFS2_Close(ADIO_File fd, int *error_code);
 void ADIOI_PVFS2_ReadContig(ADIO_File fd, void *buf, int count, 
