--- make_qamix.orig	2009-02-01 12:31:05.975310734 -0800
+++ make_qamix	2009-02-01 12:42:04.502059417 -0800
@@ -1,4 +1,4 @@
-QT_BASE_DIR=/usr/lib/qt3
+QT_BASE_DIR=$(QT_DIR)
 QT_LIB_DIR=$(QT_BASE_DIR)/lib
 QT_BIN_DIR=$(QT_BASE_DIR)/bin
 QT_INCLUDE_DIR=$(QT_BASE_DIR)/include
@@ -19,7 +19,7 @@
 	midicheckbox.o midicheckbox.moc.o \
 	midicombobox.o midicombobox.moc.o \
 	utils.o myconfig.o
-	gcc -g -o qamix \
+	$(CXX) -g -o qamix \
 	gui.o gui.moc.o parser.o \
 	hctl_element.o hctl_element.moc.o hctl_slider.o hctl_slider.moc.o \
 	hctl_checkbox.o hctl_checkbox.moc.o \
@@ -37,7 +37,7 @@
 	-L$(QT_LIB_DIR) -L$(X11_LIB_DIR) -L/usr/local/lib \
 	-lqt-mt -lasound -lm
 
-CXXFLAGS=-I$(QT_INCLUDE_DIR) -I$(X11_INCLUDE_DIR) -I. -O2 -g -Wall
+CXXFLAGS=-I$(QT_INCLUDE_DIR) -I$(X11_INCLUDE_DIR) -I. $(CFLAGS)
 
 gui.o: gui.cpp gui.h hctldata.h hctl_element.h hctl_slider.h hctl_checkbox.h hctl_combobox.h \
 	midicontrollerlist.h midicontroller.h
