Index: linux/Documentation/hwmon/c7temp
===================================================================
--- /dev/null	1970-01-01 00:00:00.000000000 +0000
+++ linux/Documentation/hwmon/c7temp	2008-06-19 15:52:02.000000000 -0700
@@ -0,0 +1,20 @@
+Kernel driver c7temp
+======================
+
+Supported chips:
+  * VIA C7
+    Prefix: 'c7temp'
+    CPUID: family 0x6, models 0xa, 0xd
+    Datasheet: Provided by VIA upon request and under NDA
+
+Authors:
+    Original work by Harald Welte and Juerg Haefliger,
+    modified by Justin Chudgar <justin@justinzane.com>
+
+Description
+-----------
+
+This driver permits reading the core temperature and voltage sensors embedded
+inside the VIA C7 CPU. Temperature is returned in millidegrees Celsius with a
+resolution of 1 degree C. Voltage is returned in millivolts with a resolution
+of 16 mV.
Index: linux/drivers/hwmon/Kconfig
===================================================================
--- linux.orig/drivers/hwmon/Kconfig	2008-06-17 17:54:58.000000000 -0700
+++ linux/drivers/hwmon/Kconfig	2008-06-19 16:06:34.000000000 -0700
@@ -730,6 +730,16 @@
 	  This driver can also be built as a module.  If so, the module
 	  will be called via686a.
 
+config SENSORS_C7TEMP
+	tristate "VIA C7 temperature sensor"
+	depends on X86 && EXPERIMENTAL
+	help
+	  If you say yes here you get support for the temperature and voltage
+	  sensors inside the VIA C7 CPU.
+
+	  This driver can also be built as a module. If so, the module will be
+	  called c7temp.
+
 config SENSORS_VT1211
 	tristate "VIA VT1211"
 	depends on EXPERIMENTAL
Index: linux/drivers/hwmon/Makefile
===================================================================
--- linux.orig/drivers/hwmon/Makefile	2008-06-17 17:57:18.000000000 -0700
+++ linux/drivers/hwmon/Makefile	2008-06-17 17:57:55.000000000 -0700
@@ -33,6 +33,7 @@
 obj-$(CONFIG_SENSORS_APPLESMC)	+= applesmc.o
 obj-$(CONFIG_SENSORS_AMS)	+= ams/
 obj-$(CONFIG_SENSORS_ATXP1)	+= atxp1.o
+obj-$(CONFIG_SENSORS_C7TEMP)	+= c7temp.o
 obj-$(CONFIG_SENSORS_CORETEMP)	+= coretemp.o
 obj-$(CONFIG_SENSORS_DME1737)	+= dme1737.o
 obj-$(CONFIG_SENSORS_DS1621)	+= ds1621.o
