diff -aur linux-3.5.7-gentoo.orig/arch/powerpc/configs/pmac32_defconfig linux-3.5.7-gentoo/arch/powerpc/configs/pmac32_defconfig
--- linux-3.5.7-gentoo.orig/arch/powerpc/configs/pmac32_defconfig	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/arch/powerpc/configs/pmac32_defconfig	2012-11-25 12:57:31.000000000 -0800
@@ -180,7 +180,7 @@
 CONFIG_ADB_CUDA=y
 CONFIG_ADB_PMU=y
 CONFIG_ADB_PMU_LED=y
-CONFIG_ADB_PMU_LED_IDE=y
+CONFIG_ADB_PMU_LED_DISK=y
 CONFIG_PMAC_APM_EMU=m
 CONFIG_PMAC_MEDIABAY=y
 CONFIG_PMAC_BACKLIGHT=y
diff -aur linux-3.5.7-gentoo.orig/arch/powerpc/configs/ppc6xx_defconfig linux-3.5.7-gentoo/arch/powerpc/configs/ppc6xx_defconfig
--- linux-3.5.7-gentoo.orig/arch/powerpc/configs/ppc6xx_defconfig	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/arch/powerpc/configs/ppc6xx_defconfig	2012-11-25 12:58:09.000000000 -0800
@@ -471,7 +471,7 @@
 CONFIG_ADB_CUDA=y
 CONFIG_ADB_PMU=y
 CONFIG_ADB_PMU_LED=y
-CONFIG_ADB_PMU_LED_IDE=y
+CONFIG_ADB_PMU_LED_DISK=y
 CONFIG_PMAC_APM_EMU=y
 CONFIG_PMAC_MEDIABAY=y
 CONFIG_PMAC_BACKLIGHT=y
diff -aur linux-3.5.7-gentoo.orig/drivers/ata/libata-core.c linux-3.5.7-gentoo/drivers/ata/libata-core.c
--- linux-3.5.7-gentoo.orig/drivers/ata/libata-core.c	2012-11-25 12:28:16.000000000 -0800
+++ linux-3.5.7-gentoo/drivers/ata/libata-core.c	2012-11-25 13:00:43.000000000 -0800
@@ -67,6 +67,7 @@
 #include <linux/cdrom.h>
 #include <linux/ratelimit.h>
 #include <linux/pm_runtime.h>
+#include <linux/leds.h>
 
 #include "libata.h"
 #include "libata-transport.h"
@@ -4834,6 +4835,9 @@
 {
 	struct ata_port *ap = qc->ap;
 
+	/* Trigger the LED (if available) */
+	ledtrig_disk_activity();
+
 	/* XXX: New EH and old EH use different mechanisms to
 	 * synchronize EH with regular execution path.
 	 *
diff -aur linux-3.5.7-gentoo.orig/drivers/ide/ide-disk.c linux-3.5.7-gentoo/drivers/ide/ide-disk.c
--- linux-3.5.7-gentoo.orig/drivers/ide/ide-disk.c	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/drivers/ide/ide-disk.c	2012-11-25 13:01:52.000000000 -0800
@@ -186,7 +186,7 @@
 	BUG_ON(drive->dev_flags & IDE_DFLAG_BLOCKED);
 	BUG_ON(rq->cmd_type != REQ_TYPE_FS);
 
-	ledtrig_ide_activity();
+	ledtrig_disk_activity();
 
 	pr_debug("%s: %sing: block=%llu, sectors=%u, buffer=0x%08lx\n",
 		 drive->name, rq_data_dir(rq) == READ ? "read" : "writ",
diff -aur linux-3.5.7-gentoo.orig/drivers/leds/Kconfig linux-3.5.7-gentoo/drivers/leds/Kconfig
--- linux-3.5.7-gentoo.orig/drivers/leds/Kconfig	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/drivers/leds/Kconfig	2012-11-25 13:04:07.000000000 -0800
@@ -443,12 +443,12 @@
 
 	  If unsure, say Y.
 
-config LEDS_TRIGGER_IDE_DISK
-	bool "LED IDE Disk Trigger"
-	depends on IDE_GD_ATA
+config LEDS_TRIGGER_DISK
+	bool "LED Disk Trigger"
+	depends on IDE_GD_ATA || ATA
 	depends on LEDS_TRIGGERS
 	help
-	  This allows LEDs to be controlled by IDE disk activity.
+	  This allows LEDs to be controlled by disk activity.
 	  If unsure, say Y.
 
 config LEDS_TRIGGER_HEARTBEAT
diff -aur linux-3.5.7-gentoo.orig/drivers/leds/ledtrig-ide-disk.c linux-3.5.7-gentoo/drivers/leds/ledtrig-ide-disk.c
--- linux-3.5.7-gentoo.orig/drivers/leds/ledtrig-ide-disk.c	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/drivers/leds/ledtrig-ide-disk.c	2012-11-25 13:10:19.000000000 -0800
@@ -18,47 +18,47 @@
 #include <linux/timer.h>
 #include <linux/leds.h>
 
-static void ledtrig_ide_timerfunc(unsigned long data);
+static void ledtrig_disk_timerfunc(unsigned long data);
 
-DEFINE_LED_TRIGGER(ledtrig_ide);
-static DEFINE_TIMER(ledtrig_ide_timer, ledtrig_ide_timerfunc, 0, 0);
-static int ide_activity;
-static int ide_lastactivity;
+DEFINE_LED_TRIGGER(ledtrig_disk);
+static DEFINE_TIMER(ledtrig_disk_timer, ledtrig_disk_timerfunc, 0, 0);
+static int disk_activity;
+static int disk_lastactivity;
 
-void ledtrig_ide_activity(void)
+void ledtrig_disk_activity(void)
 {
-	ide_activity++;
-	if (!timer_pending(&ledtrig_ide_timer))
-		mod_timer(&ledtrig_ide_timer, jiffies + msecs_to_jiffies(10));
+	disk_activity++;
+	if (!timer_pending(&ledtrig_disk_timer))
+		mod_timer(&ledtrig_disk_timer, jiffies + msecs_to_jiffies(10));
 }
-EXPORT_SYMBOL(ledtrig_ide_activity);
+EXPORT_SYMBOL(ledtrig_disk_activity);
 
-static void ledtrig_ide_timerfunc(unsigned long data)
+static void ledtrig_disk_timerfunc(unsigned long data)
 {
-	if (ide_lastactivity != ide_activity) {
-		ide_lastactivity = ide_activity;
+	if (disk_lastactivity != disk_activity) {
+		disk_lastactivity = disk_activity;
 		/* INT_MAX will set each LED to its maximum brightness */
-		led_trigger_event(ledtrig_ide, INT_MAX);
-		mod_timer(&ledtrig_ide_timer, jiffies + msecs_to_jiffies(10));
+		led_trigger_event(ledtrig_disk, INT_MAX);
+		mod_timer(&ledtrig_disk_timer, jiffies + msecs_to_jiffies(10));
 	} else {
-		led_trigger_event(ledtrig_ide, LED_OFF);
+		led_trigger_event(ledtrig_disk, LED_OFF);
 	}
 }
 
-static int __init ledtrig_ide_init(void)
+static int __init ledtrig_disk_init(void)
 {
-	led_trigger_register_simple("ide-disk", &ledtrig_ide);
+	led_trigger_register_simple("disk-activity", &ledtrig_disk);
 	return 0;
 }
 
-static void __exit ledtrig_ide_exit(void)
+static void __exit ledtrig_disk_exit(void)
 {
-	led_trigger_unregister_simple(ledtrig_ide);
+	led_trigger_unregister_simple(ledtrig_disk);
 }
 
-module_init(ledtrig_ide_init);
-module_exit(ledtrig_ide_exit);
+module_init(ledtrig_disk_init);
+module_exit(ledtrig_disk_exit);
 
 MODULE_AUTHOR("Richard Purdie <rpurdie@openedhand.com>");
-MODULE_DESCRIPTION("LED IDE Disk Activity Trigger");
+MODULE_DESCRIPTION("LED Disk Activity Trigger");
 MODULE_LICENSE("GPL");
diff -aur linux-3.5.7-gentoo.orig/drivers/leds/Makefile linux-3.5.7-gentoo/drivers/leds/Makefile
--- linux-3.5.7-gentoo.orig/drivers/leds/Makefile	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/drivers/leds/Makefile	2012-11-25 13:05:23.000000000 -0800
@@ -53,7 +53,7 @@
 
 # LED Triggers
 obj-$(CONFIG_LEDS_TRIGGER_TIMER)	+= ledtrig-timer.o
-obj-$(CONFIG_LEDS_TRIGGER_IDE_DISK)	+= ledtrig-ide-disk.o
+obj-$(CONFIG_LEDS_TRIGGER_DISK)		+= ledtrig-ide-disk.o
 obj-$(CONFIG_LEDS_TRIGGER_HEARTBEAT)	+= ledtrig-heartbeat.o
 obj-$(CONFIG_LEDS_TRIGGER_BACKLIGHT)	+= ledtrig-backlight.o
 obj-$(CONFIG_LEDS_TRIGGER_GPIO)		+= ledtrig-gpio.o
diff -aur linux-3.5.7-gentoo.orig/drivers/macintosh/Kconfig linux-3.5.7-gentoo/drivers/macintosh/Kconfig
--- linux-3.5.7-gentoo.orig/drivers/macintosh/Kconfig	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/drivers/macintosh/Kconfig	2012-11-25 13:12:12.000000000 -0800
@@ -96,18 +96,18 @@
 	  Support the front LED on Power/iBooks as a generic LED that can
 	  be triggered by any of the supported triggers. To get the
 	  behaviour of the old CONFIG_BLK_DEV_IDE_PMAC_BLINK, select this
-	  and the ide-disk LED trigger and configure appropriately through
+	  and the disk LED trigger and configure appropriately through
 	  sysfs.
 
-config ADB_PMU_LED_IDE
-	bool "Use front LED as IDE LED by default"
+config ADB_PMU_LED_DISK
+	bool "Use front LED as a disk LED by default"
 	depends on ADB_PMU_LED
 	depends on LEDS_CLASS
 	select LEDS_TRIGGERS
-	select LEDS_TRIGGER_IDE_DISK
+	select LEDS_TRIGGER_DISK
 	help
-	  This option makes the front LED default to the IDE trigger
-	  so that it blinks on IDE activity.
+	  This option makes the front LED default to the disk trigger
+	  so that it blinks on disk activity.
 
 config PMAC_SMU
 	bool "Support for SMU  based PowerMacs"
diff -aur linux-3.5.7-gentoo.orig/drivers/macintosh/via-pmu-led.c linux-3.5.7-gentoo/drivers/macintosh/via-pmu-led.c
--- linux-3.5.7-gentoo.orig/drivers/macintosh/via-pmu-led.c	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/drivers/macintosh/via-pmu-led.c	2012-11-25 13:13:08.000000000 -0800
@@ -73,8 +73,8 @@
 
 static struct led_classdev pmu_led = {
 	.name = "pmu-led::front",
-#ifdef CONFIG_ADB_PMU_LED_IDE
-	.default_trigger = "ide-disk",
+#ifdef CONFIG_ADB_PMU_LED_DISK
+	.default_trigger = "disk-activity",
 #endif
 	.brightness_set = pmu_led_set,
 };
diff -aur linux-3.5.7-gentoo.orig/include/linux/leds.h linux-3.5.7-gentoo/include/linux/leds.h
--- linux-3.5.7-gentoo.orig/include/linux/leds.h	2012-07-21 13:58:29.000000000 -0700
+++ linux-3.5.7-gentoo/include/linux/leds.h	2012-11-25 13:14:36.000000000 -0800
@@ -163,10 +163,10 @@
 #endif
 
 /* Trigger specific functions */
-#ifdef CONFIG_LEDS_TRIGGER_IDE_DISK
-extern void ledtrig_ide_activity(void);
+#ifdef CONFIG_LEDS_TRIGGER_DISK
+extern void ledtrig_disk_activity(void);
 #else
-#define ledtrig_ide_activity() do {} while(0)
+#define ledtrig_disk_activity() do {} while(0)
 #endif
 
 /*
