--- kernel/sched.c.orig	2009-02-02 08:13:43.000000000 -0800
+++ kernel/sched.c	2009-02-02 08:22:31.000000000 -0800
@@ -8898,6 +8898,14 @@
 		runtime = d->rt_runtime;
 	}
 
+	/* patch to make UID-level stuff schedulable */
+	#ifdef CONFIG_USER_SCHED
+	if (tg == &root_task_group) {
+		period = global_rt_period();
+		runtime = global_rt_runtime();
+	}
+	#endif 
+
 	/*
 	 * Cannot have more runtime than the period.
 	 */
