diff --git a/xkb/xkbAccessX.c b/xkb/xkbAccessX.c
index 7fd6a48..19f54d6 100644
--- a/xkb/xkbAccessX.c
+++ b/xkb/xkbAccessX.c
@@ -55,7 +55,7 @@ static unsigned int XkbDfltAccessXTimeoutValues = 0;
 static unsigned int XkbDfltAccessXTimeoutOptionsMask = DFLT_TIMEOUT_OPTS;
 static unsigned int XkbDfltAccessXTimeoutOptionsValues = 0;
 unsigned int XkbDfltAccessXFeedback = XkbAccessXFeedbackMask;
-unsigned short XkbDfltAccessXOptions =
+unsigned int XkbDfltAccessXOptions =
     XkbAX_AllOptionsMask & ~(XkbAX_IndicatorFBMask | XkbAX_SKReleaseFBMask |
                              XkbAX_SKRejectFBMask);
 
diff --git a/xkb/xkbInit.c b/xkb/xkbInit.c
index 30e10bd..1774f57 100644
--- a/xkb/xkbInit.c
+++ b/xkb/xkbInit.c
@@ -728,7 +728,7 @@ extern int XkbDfltRepeatInterval;
 extern unsigned short XkbDfltAccessXTimeout;
 extern unsigned int XkbDfltAccessXTimeoutMask;
 extern unsigned int XkbDfltAccessXFeedback;
-extern unsigned char XkbDfltAccessXOptions;
+extern unsigned int XkbDfltAccessXOptions;
 
 int
 XkbProcessArguments(int argc, char *argv[], int i)
@@ -789,8 +789,8 @@ XkbProcessArguments(int argc, char *argv[], int i)
                     j++;
                 }
                 if (((i + 1) < argc) && (isdigit(argv[i + 1][0]))) {
-                    XkbDfltAccessXOptions = (unsigned char)
-                        strtol(argv[++i], NULL, 16);
+                    XkbDfltAccessXOptions = (unsigned int)
+                        strtoul(argv[++i], NULL, 16);
                     j++;
                 }
             }
