--- include/servermd.h.orig	2013-09-15 15:33:09.980041174 -0700
+++ include/servermd.h	2013-09-15 15:37:24.995041296 -0700
@@ -75,13 +75,13 @@
 
 #endif                          /* __avr32__ */
 
-#ifdef __arm32__
+#if defined(__arm__) || defined(__arm32__)
 
 #define IMAGE_BYTE_ORDER        LSBFirst
 #define BITMAP_BIT_ORDER        LSBFirst
 #define GLYPHPADBYTES           4
 
-#endif                          /* __arm32__ */
+#endif                          /* __arm__ */
 
 #if defined(__nds32__)
 
