diff --git a/drivers/net/atl1c/atl1c_main.c b/drivers/net/atl1c/atl1c_main.c
index 02c7ed8..d91dabd 100644
--- a/drivers/net/atl1c/atl1c_main.c
+++ b/drivers/net/atl1c/atl1c_main.c
@@ -19,6 +19,7 @@
  * Temple Place - Suite 330, Boston, MA  02111-1307, USA.
  */
 
+#include <linux/pci-aspm.h>
 #include "atl1c.h"
 
 #define ATL1C_DRV_VERSION "1.0.1.0-NAPI"
@@ -2652,6 +2653,20 @@ static int __devinit atl1c_probe(struct pci_dev *pdev,
 
 	int err = 0;
 
+	switch (pdev->device) {
+	case PCI_DEVICE_ID_ATTANSIC_L1C:
+	case PCI_DEVICE_ID_ATTANSIC_L2C:
+	case PCI_DEVICE_ID_ATHEROS_L2C_B:
+	case PCI_DEVICE_ID_ATHEROS_L2C_B2:
+		if (pdev->subsystem_vendor == PCI_VENDOR_ID_TOSHIBA ||
+		    pdev->subsystem_vendor == PCI_VENDOR_ID_LENOVO)
+			break;
+		pci_disable_link_state(pdev, PCIE_LINK_STATE_L0S |
+				       PCIE_LINK_STATE_L1 |
+				       PCIE_LINK_STATE_CLKPM);
+		break;
+	}
+
 	/* enable device (incl. PCI PM wakeup and hotplug setup) */
 	err = pci_enable_device_mem(pdev);
 	if (err) {
-- 
1.7.7.1

  
