--- Makefile.orig	2010-01-10 14:41:29.058890209 -0800
+++ Makefile	2010-01-10 14:53:02.267904843 -0800
@@ -15,26 +15,33 @@
 
 SHELL=/bin/sh
 
-DESTDIR=
 EXECUTABLE = qemuctl
 MANPAGE = qemuctl.1
 DOCS=
 POFILES= 
 
-install: 
-	install -D $(EXECUTABLE) $(DESTDIR)$(BINDIR)/$(EXECUTABLE)
+.PHONY: targets
+
+targets: executable
+
+executable:
+	@echo "Nothing to see here...  Move along..."
+	touch $@
+
+install: targets
+	install -d $(DESTDIR)$(BINDIR)
+	install -m755 $(EXECUTABLE) $(DESTDIR)$(BINDIR)/$(EXECUTABLE)
 	install -d $(DESTDIR)$(SHAREDIR)
 	install -m644 qemuctl.xml $(DESTDIR)$(SHAREDIR)/qemuctl.xml
 	install -m644 -D $(MANPAGE) $(DESTDIR)$(MANDIR)/$(MANPAGE)
-	gzip -f --best $(DESTDIR)$(MANDIR)/$(MANPAGE)
 
 uninstall:
-	rm $(DESTDIR)$(MANDIR)/$(MANPAGE).gz
+	rm $(DESTDIR)$(MANDIR)/$(MANPAGE)
 	rm $(DESTDIR)$(BINDIR)/$(EXECUTABLE)
 	rm -rf $(DESTDIR)$(SHAREDIR)
 
 tar:
-	rm -f qemuctl$(VERSION).tar.gz
-	mkdir qemuctl$(VERSION) || true
-	cp -f $(EXECUTABLE) qemuctl.xml $(MANPAGE) Makefile qemuctl$(VERSION)
-	tar cvf qemuctl$(VERSION).tar.gz qemuctl$(VERSION)
+	rm -f qemuctl-$(VERSION).tar.gz
+	mkdir qemuctl-$(VERSION) || true
+	cp -f $(EXECUTABLE) qemuctl.xml $(MANPAGE) Makefile qemuctl-$(VERSION)
+	tar cvzf qemuctl-$(VERSION).tar.gz qemuctl-$(VERSION)
