--- qemuctl.old	2007-12-03 20:58:27.000000000 +0200
+++ qemuctl	2009-06-17 09:23:05.853962026 +0300
@@ -16,7 +16,7 @@
 # 2005 Peter Rustler - qemuctl@rustlerit.de
 # with USB-Fix from mibraun
 
-use Gtk2 qw/-init -threads-init/;
+use Gtk2 -init -threads-init;
 use Gtk2::GladeXML;
 use IO::Handle;
 use IPC::Open3;
@@ -24,6 +24,9 @@
 use Fcntl;
 use Getopt::Long qw(:config pass_through);
 
+use Gnome2;
+Gnome2::Program->init ($app_name, $version_string);
+
 $default_floppy_a = "";
 $default_floppy_b = "";
 $default_cdrom    = "";
