--- gen_initramfs.sh.orig	2018-01-09 21:42:28.002361000 -0500
+++ gen_initramfs.sh	2018-04-04 20:44:25.686227210 -0400
@@ -737,7 +737,11 @@
 		IFS=","
 		for i in ${FIRMWARE_FILES}
 		do
-			cp -L "${i}" ${TEMP}/initramfs-firmware-temp/lib/firmware/
+			if [ -d "${i}" ]; then
+				cp -a "${i}" ${TEMP}/initramfs-firmware-temp/lib/firmware/
+			else
+				cp -L "${i}" ${TEMP}/initramfs-firmware-temp/lib/firmware/
+			fi
 		done
 		IFS=$OLD_IFS
 	else
