--- gen_initramfs.sh.orig	2018-03-30 15:23:39.315703093 -0700
+++ gen_initramfs.sh	2018-03-30 15:37:26.262361760 -0700
@@ -75,6 +75,7 @@
 	mkdir -p ${TEMP}/initramfs-base-temp/run
 	mkdir -p ${TEMP}/initramfs-base-temp/sbin
 	mkdir -p ${TEMP}/initramfs-base-temp/proc
+	mkdir -p ${TEMP}/initramfs-base-temp/root
 	mkdir -p ${TEMP}/initramfs-base-temp/temp
 	mkdir -p ${TEMP}/initramfs-base-temp/tmp
 	mkdir -p ${TEMP}/initramfs-base-temp/sys
@@ -96,6 +97,10 @@
 	mknod -m 600 tty0 c 4 0 || gen_die "failed to mknod"
 	mknod -m 600 tty1 c 4 1 || gen_die "failed to mknod"
 	mknod -m 600 ttyS0 c 4 64 || gen_die "failed to mknod"
+	if [ "${SPLASH}" = 'yes' ]
+	then
+		mknod -m 600 mem c 1 1 || gen_die "failed to mknod"
+	fi
 
 	date -u '+%Y%m%d-%H%M%S' > ${TEMP}/initramfs-base-temp/etc/build_date
 	echo "Genkernel $GK_V" > ${TEMP}/initramfs-base-temp/etc/build_id
@@ -106,6 +111,12 @@
 	install -m 644 -t "${TEMP}/initramfs-base-temp/etc" /usr/share/genkernel/mdev/mdev.conf
 	install -m 755 -t "${TEMP}/initramfs-base-temp/etc/mdev/helpers" /usr/share/genkernel/mdev/helpers/storage-device
 
+	if [ "${SPLASH}" = 'yes' ]
+	then
+		cp /sbin/v86d "${TEMP}/initramfs-base-temp/sbin/"
+		chmod +x "${TEMP}/initramfs-base-temp/sbin/v86d"
+	fi
+
 	cd "${TEMP}/initramfs-base-temp/"
 	log_future_cpio_content
 	find . -print | cpio ${CPIO_ARGS} --append -F "${CPIO}" \
