--- gen_compile.sh.orig	2018-04-03 10:40:06.602524722 -0700
+++ gen_compile.sh	2018-04-03 10:53:32.335850262 -0700
@@ -327,13 +327,18 @@
 		compile_generic "${KERNEL_MAKE_DIRECTIVE_2}" kernel
 	fi
 
-	local firmware_in_kernel_line=`fgrep CONFIG_FIRMWARE_IN_KERNEL "${KERNEL_OUTPUTDIR}"/.config`
-	if [ -n "${firmware_in_kernel_line}" -a "${firmware_in_kernel_line}" != CONFIG_FIRMWARE_IN_KERNEL=y ]
+	if [ "${FIRMWARE_INSTALL}" = yes ]
 	then
-		print_info 1 "        >> Installing firmware ('make firmware_install') due to CONFIG_FIRMWARE_IN_KERNEL != y..."
-		MAKEOPTS="${MAKEOPTS} -j1" compile_generic "firmware_install" kernel
+		local firmware_in_kernel_line=`fgrep CONFIG_FIRMWARE_IN_KERNEL "${KERNEL_OUTPUTDIR}"/.config`
+		if [ -n "${firmware_in_kernel_line}" ] && [ "${firmware_in_kernel_line}" != CONFIG_FIRMWARE_IN_KERNEL=y ]
+		then
+			print_info 1 "        >> Installing firmware ('make firmware_install') due to CONFIG_FIRMWARE_IN_KERNEL != y..."
+			MAKEOPTS="${MAKEOPTS} -j1" compile_generic "firmware_install" kernel
+		else
+			print_info 1 "        >> Not installing firmware as it's included in the kernel already (CONFIG_FIRMWARE_IN_KERNEL=y)..."
+		fi
 	else
-		print_info 1 "        >> Not installing firmware as it's included in the kernel already (CONFIG_FIRMWARE_IN_KERNEL=y)..."
+		print_info 1 "        >> Not installing firmware because FIRMWARE_INSTALL does not equal yes..."
 	fi
 
 	local tmp_kernel_binary=$(find_kernel_binary ${KERNEL_BINARY_OVERRIDE:-${KERNEL_BINARY}})
