--- gen_initramfs.sh.orig	2018-04-02 12:55:04.816615455 -0700
+++ gen_initramfs.sh	2018-04-02 12:55:51.299948339 -0700
@@ -617,7 +617,11 @@
 		IFS=","
 		for i in ${FIRMWARE_FILES}
 		do
-			cp -L "${i}" ${TEMP}/initramfs-firmware-temp/lib/firmware/
+			if [ -d "${i}" ]; then
+				cp -a "${i}" ${TEMP}/initramfs-firmware-temp/lib/firmware/
+			else
+				cp -L "${i}" ${TEMP}/initramfs-firmware-temp/lib/firmware/
+			fi
 		done
 		IFS=$OLD_IFS
 	else
