--- src/common.orig	2011-04-09 17:03:55.000000000 -0700
+++ src/common	2011-04-09 17:04:38.000000000 -0700
@@ -1,20 +1,20 @@
 CFLAGS+=$(GCCFLAGS) $(INCLUDES)
-CPLUSFLAGS+=$(GCCFLAGS) $(INCLUDES)
+CXXFLAGS+=$(GCCFLAGS) $(INCLUDES)
 
 .SUFFIXES:	.C .d .cpp
 
 .C.o:
-	$(CPLUS) -c $(CPLUSFLAGS) -o $@ $<
+	$(CXX) -c $(CXXFLAGS) -o $@ $<
 
 .cpp.o:
-	$(CPLUS) -c $(CPLUSFLAGS) -o $@ $<
+	$(CXX) -c $(CXXFLAGS) -o $@ $<
 
 .c.o:
 	$(CC) -c $(CFLAGS) -o $@ $<
 
 # this one is for Visual Studio's compiler:
 %.obj:%.cpp
-	$(CPLUS) $(INCLUDES) /nologo /c /GF /EHsc $< /Fo$@
+	$(CXX) $(INCLUDES) /nologo /c /GF /EHsc $< /Fo$@
 
 %.obj:%.c
 	$(CC) $(INCLUDES) /nologo /c /GF /EHsc $< /Fo$@
--- src/solvers/Makefile.orig	2011-04-09 17:04:30.000000000 -0700
+++ src/solvers/Makefile	2011-04-09 17:23:16.000000000 -0700
@@ -3,63 +3,63 @@
 include ../config.inc
 ifneq ($(CHAFF),)
   CHAFF_SRC=sat/satcheck_zchaff.cpp sat/satcheck_zcore.cpp
-  CHAFF_INCLUDE=-I ../$(CHAFF)
-  CHAFF_LIB=../$(CHAFF)/libsat.a
+  CHAFF_INCLUDE=-I $(CHAFF)
+  CHAFF_LIB=$(CHAFF_LIBDIR)/libsat.a
 endif
 
 ifneq ($(BOOLEFORCE),)
   BOOLEFORCE_SRC=sat/satcheck_booleforce.cpp
-  BOOLEFORCE_INCLUDE=-I ../$(BOOLEFORCE)
-  BOOLEFORCE_LIB=../$(BOOLEFORCE)/libbooleforce.a
+  BOOLEFORCE_INCLUDE=-I $(BOOLEFORCE)
+  BOOLEFORCE_LIB=$(BOOLEFORCE_LIBDIR)/libbooleforce.a
 endif
 
 ifneq ($(Z3),)
   Z3_SRC=z3/z3_capi.cpp z3/z3_conv.cpp z3/z3_prop.cpp z3/z3_dec.cpp
-  Z3_INCLUDE=-I ../$(Z3)
+  Z3_INCLUDE=-I $(Z3)
   Z3_LIB=
-  CPLUSFLAGS += -DHAVE_Z3
+  CXXFLAGS += -DHAVE_Z3
 endif
 
 ifneq ($(BOOLECTOR),)
   BOOLECTOR_SRC=boolector/boolector_get.cpp \
                 boolector/boolector_prop.cpp boolector/boolector_dec.cpp
-  BOOLECTOR_INCLUDE=-I ../$(BOOLECTOR)
-  BOOLECTOR_LIB=../$(BOOLECTOR)/lib/libboolector.a
-  CPLUSFLAGS += -DHAVE_BOOLECTOR
+  BOOLECTOR_INCLUDE=-I $(BOOLECTOR)
+  BOOLECTOR_LIB=$(BOOLECTOR_LIBDIR)/libboolector.a
+  CXXFLAGS += -DHAVE_BOOLECTOR
 endif
 
 ifneq ($(MINISAT),)
   MINISAT_SRC=sat/satcheck_minisat.cpp 
-  MINISAT_INCLUDE=-I ../$(MINISAT)
-  MINISAT_LIB=../$(MINISAT)/Solver.o ../$(MINISAT)/Proof.o ../$(MINISAT)/File.o
-  CPLUSFLAGS += -DHAVE_MINISAT
+  MINISAT_INCLUDE=-I $(MINISAT)
+  MINISAT_LIB=$(MINISAT_LIBDIR)/libminisat.a
+  CXXFLAGS += -DHAVE_MINISAT
 endif
 
 ifneq ($(MINISAT2),)
   MINISAT2_SRC=sat/satcheck_minisat2.cpp
-  MINISAT2_INCLUDE=-I ../$(MINISAT2)/core -I ../$(MINISAT2)/mtl -I ../$(MINISAT2)/simp
-  MINISAT2_LIB=../$(MINISAT2)/simp/SimpSolver.o ../$(MINISAT2)/core/Solver.o
-  CPLUSFLAGS += -DHAVE_MINISAT2
+  MINISAT2_INCLUDE=-I $(MINISAT2)/core -I $(MINISAT2)/mtl -I $(MINISAT2)/simp
+  MINISAT2_LIB=$(MINISAT2_LIBDIR)/libminisat.a
+  CXXFLAGS += -DHAVE_MINISAT2
 endif
 
 ifneq ($(SMVSAT),)
   SMVSAT_SRC=sat/satcheck_smvsat.cpp
-  SMVSAT_INCLUDE=-I ../$(SMVSAT)/include
-  SMVSAT_LIB=../$(SMVSAT)/lib/libsmvsat.a
+  SMVSAT_INCLUDE=-I $(SMVSAT)/include
+  SMVSAT_LIB=$(SMVSAT_LIBDIR)/libsmvsat.a
 endif
 
 ifneq ($(SQUOLEM2),)
   SQUOLEM2_SRC=qbf/qbf_squolem.cpp qbf/qbf_squolem_core.cpp
-  SQUOLEM2_INCLUDE=-I ../$(SQUOLEM2)
-  SQUOLEM2_LIB=-L ../$(SQUOLEM2) -lsquolem2
+  SQUOLEM2_INCLUDE=-I $(SQUOLEM2)
+  SQUOLEM2_LIB=-L $(SQUOLEM2_LIBDIR) -lsquolem2
 endif
 
 ifneq ($(CUDD),)
   CUDD_SRC=qbf/qbf_bdd_core.cpp qbf/qbf_skizzo_core.cpp
-  CUDD_INCLUDE=-I ../$(CUDD)/include
-  CUDD_LIB=-L ../$(CUDD)/cudd -L ../$(CUDD)/util -L ../$(CUDD)/mtr \
-           -L ../$(CUDD)/st -L ../$(CUDD)/obj -L ../$(CUDD)/dddmp \
-           -L ../$(CUDD)/epd -lobj -lcudd -lmtr -lst -lutil -lepd -ldddmp
+  CUDD_INCLUDE=-I $(CUDD)/include
+  CUDD_LIB=-L $(CUDD_LIBDIR)/cudd -L $(CUDD_LIBDIR)/util -L $(CUDD_LIBDIR)/mtr \
+           -L $(CUDD_LIBDIR)/st -L $(CUDD_LIBDIR)/obj -L $(CUDD_LIBDIR)/dddmp \
+           -L $(CUDD_LIBDIR)/epd -lobj -lcudd -lmtr -lst -lutil -lepd -ldddmp
 endif
 
 SRC = $(CHAFF_SRC) $(BOOLEFORCE_SRC) $(MINISAT_SRC) $(MINISAT2_SRC) \
@@ -105,15 +105,15 @@
 OBJ = $(SRC:.cpp=.o)
 
 ifneq ($(SQUOLEM2),)
-  CPLUSFLAGS += -DHAVE_QBF_CORE
+  CXXFLAGS += -DHAVE_QBF_CORE
 else
 ifneq ($(CUDD),)
-  CPLUSFLAGS += -DHAVE_QBF_CORE
+  CXXFLAGS += -DHAVE_QBF_CORE
 endif
 endif
 
 ifdef MODULE_FLOATBV
-  CPLUSFLAGS += -DHAVE_FLOATBV
+  CXXFLAGS += -DHAVE_FLOATBV
   OBJ += ../floatbv/floatbv.o
 endif
 
--- src/util/Makefile.orig	2011-04-09 18:42:29.000000000 -0700
+++ src/util/Makefile	2011-04-09 18:44:21.000000000 -0700
@@ -28,7 +28,7 @@
 include ../common
 
 ifdef MODULE_FLOATBV
-  CPLUSFLAGS += -DHAVE_FLOATBV
+  CXXFLAGS += -DHAVE_FLOATBV
 endif
 
 ###############################################################################
@@ -42,7 +42,7 @@
 irep_ids.cpp: irep_ids.inc irep_ids.h
 
 irep_ids_convert: irep_ids_convert.o
-	$(CPLUS) $(LINKFLAGS) -o irep_ids_convert irep_ids_convert.o
+	$(CXX) $(LINKFLAGS) -o irep_ids_convert irep_ids_convert.o
 
 $(OBJ): irep_ids.h
 
@@ -53,4 +53,4 @@
 	rm -f *.o ieee_float_test irep_ids.h irep_ids.inc irep_ids_convert
 
 ieee_float_test: ieee_float_test.o ieee_float.o util.o
-	$(CPLUS) $(LINKFLAGS) -o ieee_float_test util.o ieee_float_test.o ../big-int/bigint.o
+	$(CXX) $(LINKFLAGS) -o ieee_float_test util.o ieee_float_test.o ../big-int/bigint.o
--- src/config.inc.orig	2011-04-09 19:15:51.000000000 -0700
+++ src/config.inc	2011-04-09 19:17:06.000000000 -0700
@@ -26,7 +26,8 @@
 #CHAFF = ../../zChaff
 #BOOLEFORCE = ../../booleforce-0.4
 #MINISAT = ../../MiniSat-p_v1.14
-MINISAT2 = ../../minisat2
+MINISAT2 = /usr/include/minisat2
+MINISAT2_LIBDIR = /usr/lib
 #SMVSAT =
 
 LANGUAGES = ansi-c csp intrep pvs smvlang verilog vhdl netlist cpp specc \
