--- src/config.inc.orig	2011-05-18 06:17:46.000000000 -0700
+++ src/config.inc	2011-07-13 10:04:46.000000000 -0700
@@ -12,7 +12,7 @@
 LIBEXT = .a
 OBJEXT = .o
 EXEEXT =
-GCCFLAGS = -Wall -MMD -O2
+GCCFLAGS = -Wall -MMD
 #LINKFLAGS = -static
 LINKLIB = ld -r -o $@ $^
 LINKBIN = $(CXX) $(LINKFLAGS) -o $@ $(LIBS)
@@ -48,7 +48,8 @@
 #CHAFF = ../../zChaff
 #BOOLEFORCE = ../../booleforce-0.4
 #MINISAT = ../../MiniSat-p_v1.14
-MINISAT2 = ../../../../minisat-2.2.0
+MINISAT2 = /usr/include/minisat2
+MINISAT2_LIBDIR = /usr/lib
 #SMVSAT =
 
 LANGUAGES = ansi-c csp intrep pvs smvlang verilog vhdl netlist cpp specc \
--- src/solvers/Makefile.orig	2011-07-13 11:17:01.000000000 -0700
+++ src/solvers/Makefile	2011-07-13 11:27:21.000000000 -0700
@@ -39,7 +39,7 @@
 ifneq ($(MINISAT2),)
   MINISAT2_SRC=sat/satcheck_minisat2.cpp
   MINISAT2_INCLUDE=-I $(MINISAT2)
-  MINISAT2_LIB=$(MINISAT2)/simp/SimpSolver$(OBJEXT) $(MINISAT2)/core/Solver$(OBJEXT)
+  MINISAT2_LIB=$(MINISAT2_LIBDIR)/libminisat$(LIBEXT)
   CXXFLAGS += -DHAVE_MINISAT2 -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS
   GCCFLAGS := $(filter-out -pedantic, $(GCCFLAGS))
 endif
