diff -ru evolution-mapi-0.32.2/src/libexchangemapi/exchange-mapi-connection.c evolution-mapi-0.32.2.new/src/libexchangemapi/exchange-mapi-connection.c
--- evolution-mapi-0.32.2/src/libexchangemapi/exchange-mapi-connection.c	2011-02-07 03:45:47.000000000 +0000
+++ evolution-mapi-0.32.2.new/src/libexchangemapi/exchange-mapi-connection.c	2011-06-09 12:44:20.287987411 +0100
@@ -2961,7 +2961,7 @@
 	}
 
 	/* Get the hierarchy table */
-	ms = GetHierarchyTable (&obj_folder, &obj_table, TableFlags_Depth | TableFlags_NoNotifications | TableFlags_UseUnicode, &row_count);
+	ms = GetHierarchyTable (&obj_folder, &obj_table, TableFlags_Depth | TableFlags_NoNotifications , &row_count);
 	if (ms != MAPI_E_SUCCESS) {
 		make_mapi_error (perror, "GetHierarchyTable", ms);
 		result = FALSE;
