=== modified file 'src/ui/clipboard.cpp'
--- src/ui/clipboard.cpp	2016-04-08 17:38:40 +0000
+++ src/ui/clipboard.cpp	2016-04-08 18:21:11 +0000
@@ -1300,7 +1300,11 @@
  */
 Glib::ustring ClipboardManagerImpl::_getBestTarget()
 {
+#if WITH_GTKMM_3_0
+    std::vector<Glib::ustring> targets = _clipboard->wait_for_targets();
+#else
     std::list<Glib::ustring> targets = _clipboard->wait_for_targets();
+#endif
 
     // clipboard target debugging snippet
     /*

