--- a/src/CMakeLists.txt	2019-11-05 11:47:43.000000000 -0800
+++ b/src/CMakeLists.txt	2019-11-05 11:48:43.000000000 -0800
@@ -253,8 +253,8 @@
         cmake_minimum_required(VERSION 3.1)
         target_compile_features(_doxygen PRIVATE cxx_alignof)
         target_compile_features(doxygen PRIVATE cxx_alignof)
-        target_compile_options(_doxygen PRIVATE -stdlib=libc++)
-        target_compile_options(doxygen PRIVATE -stdlib=libc++)
+        target_compile_options(_doxygen PRIVATE -stdlib=libstdc++)
+        target_compile_options(doxygen PRIVATE -stdlib=libstdc++)
     elseif (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
         target_compile_options(_doxygen PRIVATE -std=c++11)
         target_compile_options(doxygen PRIVATE -std=c++11)
