--- src/tclscanner.l.orig	2011-08-04 10:43:07.000000000 -0700
+++ src/tclscanner.l	2011-11-18 19:05:20.000000000 -0800
@@ -988,14 +988,21 @@
   unput('#');
   yy_push_state(COMMENTLINE);
 }
-<COMMAND>\x1A {
+<COMMAND>{ws}*\x1A {
 D
   tcl.string_commentcodify = "";
   tcl.string_commentline = "";
-  tcl.line_body1=yylineno-1;
+  tcl.line_body1=yylineno;
   tcl_command(-1,"");
 }
-<COMMAND>{ws}*[;\n] {
+<COMMAND>{ws}*; {
+D
+  tcl.string_commentcodify = "";
+  tcl.string_commentline = "";
+  tcl.line_body1=yylineno;
+  tcl_command(-1,yytext);
+}
+<COMMAND>{ws}*\n {
 D
   tcl.string_commentcodify = "";
   tcl.string_commentline = "";
