--- src/mesa/main/formats.c.orig	2012-10-28 20:03:29.000000000 -0700
+++ src/mesa/main/formats.c	2012-10-28 20:35:34.000000000 -0700
@@ -314,6 +314,15 @@
       1, 1, 2                      /* BlockWidth/Height,Bytes */
    },
    {
+      MESA_FORMAT_A16_REV,         /* Name */
+      "MESA_FORMAT_A16_REV",       /* StrName */
+      GL_ALPHA,                    /* BaseFormat */
+      GL_UNSIGNED_NORMALIZED,      /* DataType */
+      0, 0, 0, 16,                 /* Red/Green/Blue/AlphaBits */
+      0, 0, 0, 0, 0,               /* Lum/Int/Index/Depth/StencilBits */
+      1, 1, 2                      /* BlockWidth/Height,Bytes */
+   },
+   {
       MESA_FORMAT_L8,              /* Name */
       "MESA_FORMAT_L8",            /* StrName */
       GL_LUMINANCE,                /* BaseFormat */
@@ -332,6 +341,15 @@
       1, 1, 2                      /* BlockWidth/Height,Bytes */
    },
    {
+      MESA_FORMAT_L16_REV,         /* Name */
+      "MESA_FORMAT_L16_REV",       /* StrName */
+      GL_LUMINANCE,                /* BaseFormat */
+      GL_UNSIGNED_NORMALIZED,      /* DataType */
+      0, 0, 0, 0,                  /* Red/Green/Blue/AlphaBits */
+      16, 0, 0, 0, 0,              /* Lum/Int/Index/Depth/StencilBits */
+      1, 1, 2                      /* BlockWidth/Height,Bytes */
+   },
+   {
       MESA_FORMAT_I8,              /* Name */
       "MESA_FORMAT_I8",            /* StrName */
       GL_INTENSITY,                /* BaseFormat */
@@ -350,6 +368,15 @@
       1, 1, 2                      /* BlockWidth/Height,Bytes */
    },
    {
+      MESA_FORMAT_I16_REV,         /* Name */
+      "MESA_FORMAT_I16_REV",       /* StrName */
+      GL_INTENSITY,                /* BaseFormat */
+      GL_UNSIGNED_NORMALIZED,      /* DataType */
+      0, 0, 0, 0,                  /* Red/Green/Blue/AlphaBits */
+      0, 16, 0, 0, 0,              /* Lum/Int/Index/Depth/StencilBits */
+      1, 1, 2                      /* BlockWidth/Height,Bytes */
+   },
+   {
       MESA_FORMAT_YCBCR,           /* Name */
       "MESA_FORMAT_YCBCR",         /* StrName */
       GL_YCBCR_MESA,               /* BaseFormat */
@@ -404,6 +431,15 @@
       1, 1, 2
    },
    {
+      MESA_FORMAT_R16_REV,
+      "MESA_FORMAT_R16_REV",
+      GL_RED,
+      GL_UNSIGNED_NORMALIZED,
+      16, 0, 0, 0,
+      0, 0, 0, 0, 0,
+      1, 1, 2
+   },
+   {
       MESA_FORMAT_RG1616,
       "MESA_FORMAT_RG1616",
       GL_RG,
@@ -431,6 +467,15 @@
       1, 1, 4
    },
    {
+      MESA_FORMAT_ARGB2101010_REV,
+      "MESA_FORMAT_ARGB2101010_REV",
+      GL_RGBA,
+      GL_UNSIGNED_NORMALIZED,
+      10, 10, 10, 2,
+      0, 0, 0, 0, 0,
+      1, 1, 4
+   },
+   {
       MESA_FORMAT_Z24_S8,          /* Name */
       "MESA_FORMAT_Z24_S8",        /* StrName */
       GL_DEPTH_STENCIL,            /* BaseFormat */
@@ -440,6 +485,15 @@
       1, 1, 4                      /* BlockWidth/Height,Bytes */
    },
    {
+      MESA_FORMAT_Z24_S8_REV,      /* Name */
+      "MESA_FORMAT_Z24_S8_REV",    /* StrName */
+      GL_DEPTH_STENCIL,            /* BaseFormat */
+      GL_UNSIGNED_INT,             /* DataType */
+      0, 0, 0, 0,                  /* Red/Green/Blue/AlphaBits */
+      0, 0, 0, 24, 8,              /* Lum/Int/Index/Depth/StencilBits */
+      1, 1, 4                      /* BlockWidth/Height,Bytes */
+   },
+   {
       MESA_FORMAT_S8_Z24,          /* Name */
       "MESA_FORMAT_S8_Z24",        /* StrName */
       GL_DEPTH_STENCIL,            /* BaseFormat */
@@ -458,6 +512,15 @@
       1, 1, 2                      /* BlockWidth/Height,Bytes */
    },
    {
+      MESA_FORMAT_Z16_REV,         /* Name */
+      "MESA_FORMAT_Z16_REV",       /* StrName */
+      GL_DEPTH_COMPONENT,          /* BaseFormat */
+      GL_UNSIGNED_INT,             /* DataType */
+      0, 0, 0, 0,                  /* Red/Green/Blue/AlphaBits */
+      0, 0, 0, 16, 0,              /* Lum/Int/Index/Depth/StencilBits */
+      1, 1, 2                      /* BlockWidth/Height,Bytes */
+   },
+   {
       MESA_FORMAT_X8_Z24,          /* Name */
       "MESA_FORMAT_X8_Z24",        /* StrName */
       GL_DEPTH_COMPONENT,          /* BaseFormat */
@@ -476,6 +539,15 @@
       1, 1, 4                      /* BlockWidth/Height,Bytes */
    },
    {
+      MESA_FORMAT_Z24_X8_REV,      /* Name */
+      "MESA_FORMAT_Z24_X8_REV",    /* StrName */
+      GL_DEPTH_COMPONENT,          /* BaseFormat */
+      GL_UNSIGNED_INT,             /* DataType */
+      0, 0, 0, 0,                  /* Red/Green/Blue/AlphaBits */
+      0, 0, 0, 24, 0,              /* Lum/Int/Index/Depth/StencilBits */
+      1, 1, 4                      /* BlockWidth/Height,Bytes */
+   },
+   {
       MESA_FORMAT_Z32,             /* Name */
       "MESA_FORMAT_Z32",           /* StrName */
       GL_DEPTH_COMPONENT,          /* BaseFormat */
@@ -640,6 +712,15 @@
       1, 1, 16
    },
    {
+      MESA_FORMAT_RGBA_FLOAT32_REV,
+      "MESA_FORMAT_RGBA_FLOAT32_REV",
+      GL_RGBA,
+      GL_FLOAT,
+      32, 32, 32, 32,
+      0, 0, 0, 0, 0,
+      1, 1, 16
+   },
+   {
       MESA_FORMAT_RGBA_FLOAT16,
       "MESA_FORMAT_RGBA_FLOAT16",
       GL_RGBA,
@@ -1316,6 +1397,15 @@
       1, 1, 8
    },
    {
+      MESA_FORMAT_RGBA_16_REV,
+      "MESA_FORMAT_RGBA_16_REV",
+      GL_RGBA,
+      GL_UNSIGNED_NORMALIZED,
+      16, 16, 16, 16,
+      0, 0, 0, 0, 0,
+      1, 1, 8
+   },
+   {
      MESA_FORMAT_RED_RGTC1,
      "MESA_FORMAT_RED_RGTC1",
      GL_RED,
@@ -2097,6 +2187,7 @@
       return;
 
    case MESA_FORMAT_ARGB2101010:
+   case MESA_FORMAT_ARGB2101010_REV:
       *datatype = GL_UNSIGNED_INT_2_10_10_10_REV;
       *comps = 4;
       return;
@@ -2128,9 +2219,13 @@
       return;
 
    case MESA_FORMAT_R16:
+   case MESA_FORMAT_R16_REV:
    case MESA_FORMAT_A16:
+   case MESA_FORMAT_A16_REV:
    case MESA_FORMAT_L16:
+   case MESA_FORMAT_L16_REV:
    case MESA_FORMAT_I16:
+   case MESA_FORMAT_I16_REV:
       *datatype = GL_UNSIGNED_SHORT;
       *comps = 1;
       return;
@@ -2156,6 +2251,7 @@
       return;
 
    case MESA_FORMAT_Z24_S8:
+   case MESA_FORMAT_Z24_S8_REV:
       *datatype = GL_UNSIGNED_INT_24_8_MESA;
       *comps = 2;
       return;
@@ -2166,6 +2262,7 @@
       return;
 
    case MESA_FORMAT_Z16:
+   case MESA_FORMAT_Z16_REV:
       *datatype = GL_UNSIGNED_SHORT;
       *comps = 1;
       return;
@@ -2176,6 +2273,7 @@
       return;
 
    case MESA_FORMAT_Z24_X8:
+   case MESA_FORMAT_Z24_X8_REV:
       *datatype = GL_UNSIGNED_INT;
       *comps = 1;
       return;
@@ -2220,6 +2318,7 @@
       return;
 
    case MESA_FORMAT_RGBA_16:
+   case MESA_FORMAT_RGBA_16_REV:
       *datatype = GL_UNSIGNED_SHORT;
       *comps = 4;
       return;
@@ -2697,18 +2796,23 @@
    case MESA_FORMAT_A8:
       return format == GL_ALPHA && type == GL_UNSIGNED_BYTE;
    case MESA_FORMAT_A16:
-      return format == GL_ALPHA && type == GL_UNSIGNED_SHORT &&
-         littleEndian && !swapBytes;
+      return format == GL_ALPHA && type == GL_UNSIGNED_SHORT;
+   case MESA_FORMAT_A16_REV:
+      return GL_FALSE;
    case MESA_FORMAT_L8:
       return format == GL_LUMINANCE && type == GL_UNSIGNED_BYTE;
    case MESA_FORMAT_L16:
+      return format == GL_LUMINANCE && type == GL_UNSIGNED_SHORT;
+   case MESA_FORMAT_L16_REV:
+      return GL_FALSE;
       return format == GL_LUMINANCE && type == GL_UNSIGNED_SHORT &&
          littleEndian && !swapBytes;
    case MESA_FORMAT_I8:
       return format == GL_INTENSITY && type == GL_UNSIGNED_BYTE;
    case MESA_FORMAT_I16:
-      return format == GL_INTENSITY && type == GL_UNSIGNED_SHORT &&
-         littleEndian && !swapBytes;
+      return format == GL_INTENSITY && type == GL_UNSIGNED_SHORT;
+   case MESA_FORMAT_I16_REV:
+      return GL_FALSE;
 
    case MESA_FORMAT_YCBCR:
    case MESA_FORMAT_YCBCR_REV:
@@ -2722,8 +2826,9 @@
       return GL_FALSE;
 
    case MESA_FORMAT_R16:
-      return format == GL_RED && type == GL_UNSIGNED_SHORT && littleEndian &&
-         !swapBytes;
+      return format == GL_RED && type == GL_UNSIGNED_SHORT;
+   case MESA_FORMAT_R16_REV:
+      return GL_FALSE;
    case MESA_FORMAT_RG1616:
       return format == GL_RG && type == GL_UNSIGNED_SHORT && littleEndian &&
          !swapBytes;
@@ -2731,19 +2836,23 @@
       return GL_FALSE;
 
    case MESA_FORMAT_ARGB2101010:
-      return format == GL_BGRA && type == GL_UNSIGNED_INT_2_10_10_10_REV &&
-         !swapBytes;
+      return format == GL_BGRA && type == GL_UNSIGNED_INT_2_10_10_10_REV && littleEndian;
+   case MESA_FORMAT_ARGB2101010_REV:
+      return format == GL_BGRA && type == GL_UNSIGNED_INT_10_10_10_2 && !littleEndian;
 
    case MESA_FORMAT_Z24_S8:
-      return format == GL_DEPTH_STENCIL && type == GL_UNSIGNED_INT_24_8 &&
-         !swapBytes;
+      return format == GL_DEPTH_STENCIL && type == GL_UNSIGNED_INT_24_8;
+   case MESA_FORMAT_Z24_S8_REV:
+       return format == GL_DEPTH_STENCIL && type == GL_UNSIGNED_INT_8_24_REV_MESA;
    case MESA_FORMAT_Z24_X8:
+   case MESA_FORMAT_Z24_X8_REV:
    case MESA_FORMAT_S8_Z24:
       return GL_FALSE;
 
    case MESA_FORMAT_Z16:
-      return format == GL_DEPTH_COMPONENT && type == GL_UNSIGNED_SHORT &&
-         !swapBytes;
+      return format == GL_DEPTH_COMPONENT && type == GL_UNSIGNED_SHORT;
+   case MESA_FORMAT_Z16_REV:
+      return GL_FALSE;
 
    case MESA_FORMAT_X8_Z24:
       return GL_FALSE;
@@ -2776,6 +2885,8 @@
 
    case MESA_FORMAT_RGBA_FLOAT32:
       return format == GL_RGBA && type == GL_FLOAT && !swapBytes;
+   case MESA_FORMAT_RGBA_FLOAT32_REV:
+      return GL_FALSE;
    case MESA_FORMAT_RGBA_FLOAT16:
       return format == GL_RGBA && type == GL_HALF_FLOAT && !swapBytes;
 
@@ -2904,6 +3015,7 @@
    case MESA_FORMAT_SIGNED_RGB_16:
    case MESA_FORMAT_SIGNED_RGBA_16:
    case MESA_FORMAT_RGBA_16:
+   case MESA_FORMAT_RGBA_16_REV:
       /* FINISHME: SNORM */
       return GL_FALSE;
 
