--- CMakeLists.txt.orig	2017-10-31 15:09:18.856113148 -0700
+++ CMakeLists.txt	2017-10-31 15:12:16.538305005 -0700
@@ -29,7 +29,7 @@
 pkg_check_modules(LIBXMLPP libxml++-2.6)
 pkg_check_modules(LIBCURL libcurl)
 
-set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++0x -Wall -D_GLIBCXX_USE_NANOSLEEP -DPACKAGE=emilpro -DPACKAGE_VERSION=1")
+set (CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -D_GLIBCXX_USE_NANOSLEEP -DPACKAGE=emilpro -DPACKAGE_VERSION=1 -lpthread" )
 LINK_DIRECTORIES(${CMAKE_BINARY_DIR}/install-binutils/lib ${CMAKE_BINARY_DIR}/install-binutils/lib64)
 
 if (HAVE_BFD_MULTIARCH)
