--- configure.in.orig	2016-04-29 11:06:47.642487076 -0700
+++ configure.in	2016-04-29 11:14:12.959680789 -0700
@@ -6893,6 +6893,9 @@
 fi
 AC_SUBST(MOZ_GL_DEFAULT_PROVIDER)
 AC_DEFINE_UNQUOTED(GL_PROVIDER_$MOZ_GL_DEFAULT_PROVIDER)
+if test "$MOZ_GL_PROVIDER" = "EGL"; then
+    AC_DEFINE(MOZ_EGL_XRENDER_COMPOSITE)
+fi
 
 dnl ========================================================
 dnl = faststripe theme
--- configure.orig	2016-04-29 11:06:39.915580500 -0700
+++ configure	2016-04-29 11:15:07.184703614 -0700
@@ -25428,6 +25428,9 @@
   withval="$with_gl_provider"
    val=`echo $withval`
     MOZ_GL_PROVIDER="$val"
+    if test "$val" = "EGL"; then
+      MOZ_EGL_XRENDER_COMPOSITE=1
+    fi
 fi
 
 
--- gfx/gl/GLContextProviderImpl.h.orig	2016-04-29 16:46:12.009987108 -0700
+++ gfx/gl/GLContextProviderImpl.h	2016-04-29 17:36:38.108466759 -0700
@@ -10,7 +10,7 @@
 #ifndef GL_CONTEXT_PROVIDER_NAME
 #error GL_CONTEXT_PROVIDER_NAME not defined
 #endif
-#if defined(ANDROID)
+#if defined(ANDROID) || defined(GL_PROVIDER_EGL)
 typedef void* EGLSurface;
 #endif // defined(ANDROID)
 
@@ -84,7 +84,7 @@
     static already_AddRefed<GLContext>
     CreateWrappingExisting(void* aContext, void* aSurface);
 
-#if defined(ANDROID)
+#if defined(ANDROID) || defined(GL_PROVIDER_EGL)
     static EGLSurface CreateEGLSurface(void* aWindow);
     static void DestroyEGLSurface(EGLSurface surface);
 #endif // defined(ANDROID)
--- gfx/gl/GLContextProvider.h.orig	2016-04-29 20:51:45.350686842 -0700
+++ gfx/gl/GLContextProvider.h	2016-04-29 20:58:03.616317703 -0700
@@ -40,35 +40,28 @@
 #endif
 
 #if defined(MOZ_X11)
+  #ifdef MOZ_EGL_XRENDER_COMPOSITE
+    #define GL_CONTEXT_PROVIDER_NAME GLContextProviderEGL
+    #include "GLContextProviderImpl.h"
+    #undef GL_CONTEXT_PROVIDER_NAME
+    #define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderEGL
+  #endif
+#else
   #define GL_CONTEXT_PROVIDER_NAME GLContextProviderGLX
   #include "GLContextProviderImpl.h"
   #undef GL_CONTEXT_PROVIDER_NAME
   #define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderGLX
 #endif
 
-#define GL_CONTEXT_PROVIDER_NAME GLContextProviderEGL
-#include "GLContextProviderImpl.h"
-#undef GL_CONTEXT_PROVIDER_NAME
-#ifndef GL_CONTEXT_PROVIDER_DEFAULT
-  #define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderEGL
-#endif
-
 #if defined(MOZ_WIDGET_UIKIT)
 #define GL_CONTEXT_PROVIDER_NAME GLContextProviderEAGL
 #include "GLContextProviderImpl.h"
 #undef GL_CONTEXT_PROVIDER_NAME
 #ifndef GL_CONTEXT_PROVIDER_DEFAULT
-#define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderEAGL
+  #define GL_CONTEXT_PROVIDER_DEFAULT GLContextProviderEAGL
 #endif
 #endif
 
-#ifdef MOZ_GL_PROVIDER
-  #define GL_CONTEXT_PROVIDER_NAME MOZ_GL_PROVIDER
-  #include "GLContextProviderImpl.h"
-  #undef GL_CONTEXT_PROVIDER_NAME
-  #define GL_CONTEXT_PROVIDER_DEFAULT MOZ_GL_PROVIDER
-#endif
-
 #ifdef GL_CONTEXT_PROVIDER_DEFAULT
   typedef GL_CONTEXT_PROVIDER_DEFAULT GLContextProvider;
 #else
