--- a/build/autoconf/compiler-opts.m4	2017-03-27 13:05:30.237437847 -0700
+++ b/build/autoconf/compiler-opts.m4	2017-03-27 13:07:33.848963990 -0700
@@ -288,6 +288,8 @@
 
 if test "$GNU_CC" -a -n "$MOZ_PIE"; then
     AC_MSG_CHECKING([for PIE support])
+    _SAVE_CFLAGS=$CFLAGS
+    CFLAGS="$CFLAGS -fPIC"
     _SAVE_LDFLAGS=$LDFLAGS
     LDFLAGS="$LDFLAGS -pie"
     AC_TRY_LINK(,,AC_MSG_RESULT([yes])
@@ -295,6 +297,7 @@
                   AC_MSG_RESULT([no])
                   AC_MSG_ERROR([--enable-pie requires PIE support from the linker.]))
     LDFLAGS=$_SAVE_LDFLAGS
+    CFLAGS=$_SAVE_CFLAGS
 fi
 
 AC_SUBST(MOZ_PROGRAM_LDFLAGS)
