--- media/webrtc/trunk/src/modules/PCM16B.target.mk.orig	2012-12-26 12:00:52.417668321 -0800
+++ media/webrtc/trunk/src/modules/PCM16B.target.mk	2012-12-26 12:04:06.767687709 -0800
@@ -54,6 +54,7 @@
 
 INCS_Debug := -Isrc \
 	-I. \
+	-Isrc/common_audio/signal_processing/include
 	-Isrc/modules/audio_coding/codecs/pcm16b/include
 
 DEFS_Release := '-D_FILE_OFFSET_BITS=64' \
@@ -110,6 +111,7 @@
 
 INCS_Release := -Isrc \
 	-I. \
+	-Isrc/common_audio/signal_processing/include
 	-Isrc/modules/audio_coding/codecs/pcm16b/include
 
 OBJS := $(obj).target/$(TARGET)/src/modules/audio_coding/codecs/pcm16b/pcm16b.o
--- media/webrtc/trunk/src/modules/audio_coding/codecs/pcm16b/pcm16b.gypi.orig	2012-12-26 18:03:46.759840439 -0800
+++ media/webrtc/trunk/src/modules/audio_coding/codecs/pcm16b/pcm16b.gypi	2012-12-26 18:05:41.713185240 -0800
@@ -11,6 +11,9 @@
     {
       'target_name': 'PCM16B',
       'type': '<(library)',
+      'dependencies': [
+        '<(webrtc_root)/common_audio/common_audio.gyp:signal_processing',
+      ],
       'include_dirs': [
         'include',
       ],
