--- gtk/gtkmm/recentinfo.cc~	2009-01-20 17:33:35.000000000 +0100
+++ gtk/gtkmm/recentinfo.cc	2009-08-20 01:22:56.662814220 +0200
@@ -189,7 +189,7 @@
 
 bool RecentInfo::get_application_info(const Glib::ustring& app_name, Glib::StringArrayHandle& app_exec, guint& count, time_t& time) const
 {
-  return gtk_recent_info_get_application_info(const_cast<GtkRecentInfo*>(gobj()), app_name.c_str(), const_cast<gchar**>((app_exec).data()), &(count), &(time));
+  return gtk_recent_info_get_application_info(const_cast<GtkRecentInfo*>(gobj()), app_name.c_str(), const_cast<const gchar**>((app_exec).data()), &(count), &(time));
 }
 
 Glib::ustring RecentInfo::last_application() const
