--- Source/cmake/OptionsCommon.cmake.orig	2015-07-18 14:57:36.719795980 -0700
+++ Source/cmake/OptionsCommon.cmake	2015-07-18 14:59:31.359795952 -0700
@@ -1,5 +1,7 @@
 add_definitions(-DBUILDING_WITH_CMAKE=1)
 add_definitions(-DHAVE_CONFIG_H=1)
+add_definitions(-DWTF_USE_OPENGL=1)
+add_definitions(-DWTF_USE_GLX=0)
 
 # CODE_GENERATOR_PREPROCESSOR_WITH_LINEMARKERS only matters with GCC >= 4.7.0.  Since this
 # version, -P does not output empty lines, which currently breaks make_names.pl in
