--- Source/cmake/OptionsGTK.cmake.orig	2015-12-06 21:33:08.013265782 -0800
+++ Source/cmake/OptionsGTK.cmake	2015-12-06 21:46:55.703152185 -0800
@@ -60,10 +60,10 @@
 # the feature is by default always on (ENABLE_OPENGL=ON).
 # What we select here automatically is if we use OPENGL (ENABLE_GLES2=OFF)
 # or OPENGLES2 (ENABLE_GLES2=ON) for building the feature.
-set(ENABLE_GLES2_DEFAULT OFF)
+set(ENABLE_GLES2_DEFAULT ON)
 
-if (NOT OPENGL_FOUND AND OPENGLES2_FOUND)
-    set(ENABLE_GLES2_DEFAULT ON)
+if (NOT OPENGLES2_FOUND)
+    set(ENABLE_GLES2_DEFAULT OFF)
 endif ()
 
 WEBKIT_OPTION_DEFINE(ENABLE_GLES2 "Whether to enable OpenGL ES 2.0." PUBLIC ${ENABLE_GLES2_DEFAULT})
@@ -233,6 +233,9 @@
     if (EGL_FOUND)
         list(APPEND CAIROGL_COMPONENTS cairo-egl)
     endif ()
+    if (GLES2_FOUND)
+        list(APPEND CAIROGL_COMPONENTS cairo-glesv2)
+    endif ()
 
     find_package(CairoGL 1.10.2 COMPONENTS ${CAIROGL_COMPONENTS})
     if (NOT CAIROGL_FOUND)
